pkg/frontend/querymiddleware/codec.go

```diff
@@ -409,6 +496,53 @@ func (c prometheusCodec) EncodeRequest(ctx context.Context, r Request) (*http.Re
 	return req.WithContext(ctx), nil
 }
 
+func (c prometheusCodec) EncodeLabelsQueryRequest(ctx context.Context, req LabelsQueryRequest) (*http.Request, error) {
+	var u *url.URL
+	switch req := req.(type) {
+	case *PrometheusLabelNamesQueryRequest:
```

Comment:
I don't understand why this can only encode one of the implementations of `LabelsQueryRequest`

Reply:
whoop yeah missed that as I originally only intended to PR for the Label Names but ended up adding Label Values as well... coming up

Reply:
added, with test cases